src/sentry/notifications/notifications/codeowners_auto_sync.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Iterable, Mapping, MutableMapping
+
+from sentry.notifications.notifications.base import ProjectNotification
+from sentry.notifications.types import NotificationSettingTypes
+from sentry.types.integrations import ExternalProviders
+from sentry.utils.http import absolute_uri
+
+if TYPE_CHECKING:
+    from sentry.models import Team, User
+
+
+class AutoSyncNotification(ProjectNotification):
+    notification_setting_type = NotificationSettingTypes.DEPLOY
+
+    def determine_recipients(self) -> Iterable[Team | User]:
+        raise self.organization.get_owners()
+
+    def get_participants(self) -> Mapping[ExternalProviders, Iterable[Team | User]]:
+        # For now, filter to only email.
+        return {ExternalProviders.EMAIL: super().get_participants()[ExternalProviders.EMAIL]}
```

Comment:
@NisanthanNanthakumar I think you can handle doing this email-only more elegantly with `get_notification_providers` and have that just return email. The downside of doing it this way is you're actually going to calculate what we'd want for Slack (which performs some queries) and just discard that result.